runtimes/aleph-debian-11-python/init1.py

```diff
@@ -107,35 +109,52 @@ def setup_variables(variables: Optional[Dict[str, str]]):
 
 
 def setup_network(
-    ip: Optional[str], route: Optional[str], dns_servers: Optional[List[str]] = None
+    ip: Optional[str],
+    ipv6: Optional[str],
+    route: Optional[str],
+    ipv6_gateway: Optional[str],
+    dns_servers: Optional[List[str]] = None,
 ):
     """Setup the system with info from the host."""
     dns_servers = dns_servers or []
     if not os.path.exists("/sys/class/net/eth0"):
         logger.info("No network interface eth0")
         return
 
-    if not ip:
+    if not (ip or ipv6):
         logger.info("No network IP")
         return
 
-    logger.debug("Setting up networking")
-    system("ip addr add 127.0.0.1/8 dev lo brd + scope host")
-    system("ip addr add ::1/128 dev lo")
-    system("ip link set lo up")
-    if "/" in ip:
+    if ip:
+        logger.debug("Setting up IPv4")
+        system("ip addr add 127.0.0.1/8 dev lo brd + scope host")
+        system("ip addr add ::1/128 dev lo")
+        system("ip link set lo up")
+        system(f"ip addr add {ip} dev eth0")
+
         # Forward compatibility with future supervisors that pass the mask with the IP.
+        if "/" not in ip:
+            logger.warning(
+                "Not passing the mask with the IP is deprecated and will be unsupported"
+            )
+            ip = f"{ip}/24"
+
         system(f"ip addr add {ip} dev eth0")
-    else:
-        logger.warning("Not passing the mask with the IP is deprecated and will be unsupported")
-        system(f"ip addr add {ip}/24 dev eth0")
-    system("ip link set eth0 up")
+        system("ip link set eth0 up")
 
     if route:
```

Comment:
Because I wasn't sure it was the right place. I wanted to enable the interface only after all addresses were set up, so `if ip` is not necessarily the best place. The only "problem" here is that the interface will be up even if the VM does not support networking, but that seems like a minor issue.

Reply:
Solved in your PR, marking this as resolved.